### ISSUE TITLE
Implement backup list display functionality

### ISSUE THREAD

**Issue by kengo-k** [kengo-k/taxdesk#26]
## Purpose
Implement a functional backup list display in the web UI to allow users to view available backups and make informed restoration decisions based on migration compatibility.

## Requirements
- Implement backup list API that returns backup information with migration numbers
- Create UI to display backup list with migration numbers
- Show current migration version alongside backup migration versions

### API Implementation
- **Endpoint**: `/api/backup` (GET)
- **Backend**: Use existing `listBackups()` function from `backup-lib.js`
- **Response**: List of backups with migration information
- **Integration**: Call CLI library functions directly (no external command execution)

### Frontend Implementation
- **Current Migration Display**: Call `/api/migration-version` (from #24) to show current version
- **Backup List Display**: Show each backup's migration version
- **UI Layout**: Exact placement to be determined during implementation

## Implementation Approach
- Leverage existing CLI implementation (`backup-lib.js`)
- Use `listBackups()` function which already returns migration information
- Integrate with migration version API from #24 for current version display

## Prerequisites
Depends on #24 (Migration version API implementation)

## Implementation Scope
- ✅ Backup list API implementation
- ✅ UI for backup list with migration information
- ❌ Backup creation functionality (separate issue)
- ❌ Restore functionality (separate issue)
